spec/views/recipes/recipe_show_spec.rb

```diff
@@ -0,0 +1,48 @@
+require 'rails_helper'
+
+RSpec.describe 'recipes/show', type: :view do
+  before(:each) do
+    @user = FactoryBot.create(:user)
+    sign_in @user
+
+    @recipe = FactoryBot.create(:recipe, user: @user)
+
+    @recipe_foods = [FactoryBot.create(:recipe_food, recipe: @recipe)]
+  end
+
+  it 'renders the recipe details' do
+    render
+
+    expect(rendered).to have_content @recipe.name
+    expect(rendered).to have_content "#{@recipe.preparation_time} minutes"
+    expect(rendered).to have_content "#{@recipe.cooking_time} minutes"
+    expect(rendered).to have_content @recipe.description
+    expect(rendered).to have_content 'Public:'
+
+    if @recipe.user == @user
+      expect(rendered).to have_button('Yes')
+    else
+      expect(rendered).to have_button('No')
+    end
+
+    if @recipe.user == @user
+      expect(rendered).to have_link('Add New Food')
+    else
+      expect(rendered).not_to have_link('Add New Food')
+    end
+
+    if @recipe.is_public || @recipe.user == @user
+      expect(rendered).to have_link('Generate shopping list')
+    else
+      expect(rendered).not_to have_link('Generate shopping list')
+    end
+
```

Comment:
- [ ] Kindly note that one of your tests is not passing kindly work on that add make sure all the test pass

![image](https://github.com/jlvFlores/RoR-recipe-app/assets/65068771/f544f978-6449-4178-9b45-9f79623eb7c4)